OpenTrons/opentrons-api
# refactor(api): Allow homing after drop tip in engine core

# Overview

closes https://opentrons.atlassian.net/browse/RCORE-527.
Changed drop_tip, home_after arg to optional and added a default value to None so that the hardware controller can decide depends on the hardware if it should home or not.

# Test Plan

- Upload the following protocol to an OT-2 with core ff on and off. 
- Make sure the pipette homes if `home_after` arg is not specified in the protocol `drop_tip`.
- Make sure the pipette does not home if setting `home_after=False`. 

```
from opentrons import protocol_api, types

metadata = {
    "apiLevel": "2.12",
}

def run(ctx: protocol_api.ProtocolContext) -> None:
    tip_rack = ctx.load_labware("opentrons_96_tiprack_300ul", 6)
    pipette = ctx.load_instrument("p20_single_GEN2", types.Mount.RIGHT, [tip_rack])
    well = tip_rack.wells()[0]

    pipette.pick_up_tip(well.top())
    pipette.drop_tip()

```

# Changelog

- Changed public instrument context drop_tip and return top to accept home_after as an Optional arg.
- Added `homeAfter` prop to `DropTipParams`.
- Wired up `core/engine` `drop_tip` with `home_after`
- Added a `home_after` default value to hardware_stopper

# Review requests

Does changes make sense? did I forget anything? 

# Risk assessment

Low. Should not effect the logic bc we are still setting to True in case its not specified.
